### ISSUE TITLE
Weird focus-event related problem after upgraded to 0.47.0

### ISSUE THREAD

**Issue by philiptzou** [Hacker0x01/react-datepicker#885]
Hi everyone. I just encountered a weird focus-event related problem after I upgraded the `react-datepicker` to 0.47.0. I tested the following steps on both Chrome and Firefox and it seems both browsers get the same problem. Here are the simple steps to reproduce the problem:

1. Open https://hacker0x01.github.io/react-datepicker/
2. Scroll down to "Default" example. Left click the datepicker input box, then select a random date.
3. Press "tab" key on the keyboard.

Then you will see the browser enters a dead loop and keeps switching between the two datepicker input box.

The problem is gone after downgraded to 0.46.0.

Any idea why this problem happened? I checked the [comparison between 0.46.0 and 0.47.0](https://github.com/Hacker0x01/react-datepicker/compare/v0.46.0...v0.47.0) but didn't find anything interesting.

**Comment by alex-shamshurin** [comment]
Really weird. Looks like with wrong `ref` or `key` somewhere.

**Comment by conlanpatrek** [comment]
I don't know that it's an incorrect ref...

It looks like an issue the `deferFocusInput` call (https://github.com/Hacker0x01/react-datepicker/blob/master/src/datepicker.jsx#L175). When you blur from input A (with an open datepicker) by focusing on input B, you set a timeout that refocuses input A, but by the time that cb gets called, input B's datepicker is popped open. So now, when input B blurs, it will call `deferFocusInput` which primes it to steal focus back from input A. Wash, rinse, repeat until infinity.

My guess is that this never got noticed before because you're also closing the date picker on mouse down somewhere (probably in `react-onclickoutside`), so clicking into a second datepicker doesn't trigger the loop.

**Comment by conlanpatrek** [comment]
I should also mention that I can recreate the issue with 0.46.0

**Comment by philiptzou** [comment]
@conlanpatrek Really? 0.46.0 works fine on my computers. Are there any different steps you used?

**Comment by philiptzou** [comment]
I think I found the reason.

In version 0.46.0, [`datepicker.jsx`](https://github.com/Hacker0x01/react-datepicker/blob/v0.46.0/src/datepicker.jsx#L265-L267) has following three lines:

```javascript
      if (/^Arrow/.test(event.key)) {
        this.onInputClick()
      }
```

This behaviour was changed by https://github.com/Hacker0x01/react-datepicker/commit/b1decffb4bbeabed8b0313050f0f57acc49e6208#diff-f662ae56197d5c888897a02b982f0ad3. Instead of testing if `event.key` was one of the `ArrowUp`, `ArrowDown`, `ArrowLeft` and `ArrowRight`, the new version captured all key event which include `tab`.

The `onInputClick()` opened the calendar panel by setting `this.state.open` to `true`. However, the `tab` also triggered `blur` event. In `handleBlur()`, if `this.state.open` was `true`, `deferFocusInput()` would be called and eventually make the blurred `input` being focused again.

Then the following dead loop was described precisely by @conlanpatrek .
